graylog2-server/src/main/java/org/graylog/plugins/views/migrations/V20200204122000_MigrateUntypedViewsToDashboards/Query.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.migrations.V20200204122000_MigrateUntypedViewsToDashboards;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.graph.Traverser;
+import org.bson.Document;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static java.util.stream.Collectors.toSet;
+
+class Query {
+    private static final String FIELD_ID = "id";
+    private static final String FIELD_FILTER = "filter";
+    private static final String FIELD_SEARCH_TYPE_ID = "id";
+    private static final String FIELD_SEARCH_TYPES = "search_types";
+    private static final String FIELD_QUERY = "query";
+    private static final String FIELD_QUERY_STRING = "query_string";
+    private static final String FIELD_TIMERANGE = "timerange";
+    private static final String FIELD_SUB_FILTERS = "filters";
+    private static final String FIELD_FILTER_TYPE = "type";
+    private static final String TYPE_STREAM_FILTER = "stream";
+    private static final String FIELD_STREAM_ID = "id";
+
+    private final Document queryDocument;
+
+    Query(Document queryDocument) {
+        this.queryDocument = queryDocument;
+    }
+
+    String id() {
+        return this.queryDocument.getString(FIELD_ID);
+    }
+
+    private Set<SearchType> searchTypesByIds(Set<String> searchTypeIds) {
```

Comment:
Maybe it's just me, but I think `searchTypesByIds` implies a map keyed by ID and would prefer the name `searchTypesForIds`.

Reply:
✅